x-pack/platform/plugins/shared/lens/server/content_management/lens_storage.ts

```diff
@@ -136,6 +136,9 @@ export class LensStorage extends SOContentStorage<LensCrudTypes> {
     // Save data in DB
     const soClient = await savedObjectClientFromRequest(ctx);
 
+    // since we use create below this is to throw if SO id not found
+    await soClient.get(CONTENT_ID, id);
```

Comment:
Without this addition, the update call will create the SO even if it does not already exist.